server/src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -525,8 +524,19 @@ private String createDN(Consumer consumer, Owner owner) {
         containerSet.add(container);
         container.setId("content_access");
         container.setName(" Content Access");
-        for (Content c : ownerContent) {
-            container.addContent(c, false);
+        Set<String> contentUuids = new HashSet<>();
+
+        for (Product product : ownerProduct) {
+            for (ProductContent pc : product.getProductContent()) {
+                if (contentUuids.contains(pc.getContent().getUuid())) {
```

Comment:
I was not sure what should be the expected behavior with mix content enablement, I have already asked PM's about it
and waiting for a reply. Modified the code to make sure that the "enabled" version win out.
I noticed that content can exist without product, when we query owner-content we get unwanted content, so decided to go via owner-product (and ideally it should be content of product with active pool, there was some work in that area with ENT-2352, not sure if it is there in master branch)

Reply:
In non-SCA mode, we simply return duplicates (or more) of the same content that might be enabled or disabled, but that's because they come from different entitlement certs. I wonder what happens on the client side in that case (the client tools guys can help here). Do we actually end up with duplicate entries in redhat.repo?

Reply:
@cnsnyder @jirihnidek ^^
I think I recall seeing discussion that it ends up being that if duplicates exist, then the first found is what is used; which is probably not desired overall (undefined behavior is bad behavior). If that's the case, then I think we should get some PMs involved, figure out what they expect, and then determine where the new behavior should be. I expect that it'll mostly fall to CP to generate something a tad more sane, since subman is already implementing a contingency plan, even if that implementation isn't optimal or deterministic.

Reply:
Also, regardless, this branch is a bit special, since everything is being added to a single product. That in itself is causing deduplication work done elsewhere to keep the last update received (Product.addContent -> Product.addProductContent).

So really, the ask here is what PMs expect in the case of multiple entitlements providing a given content in different states of enablement.